### PR TITLE
chore: migrate eslint to flat config

### PR DESCRIPTION
**Pull request by BBboy01**
use `npx eslint --inspect-config` browse all configured rules

https://github.com/KillerCodeMonkey/ngx-quill/blob/db73b0c2f394a0cd47734fd1af2165fc33394f3e/.eslintrc.json#L3-L5

previous ignore pattern ignored all ts files, eslint seems lint nothing

- remove unused plugins `eslint-plugin-prefer-arrow` and `eslint-plugin-import`
- replace `@angular-eslint/*` with `angular-eslint`
- `'@typescript-eslint/no-explicit-any': 'off'` as rule

### CODE REVIEW

File: projects/ngx-quill/src/lib/quill-editor.component.spec.ts [hunk 1]
```diff
@@ -1355,7 +1355,7 @@ describe('QuillEditor - base config', () => {
       jasmine.objectContaining({attrName: 'size', keyName: 'font-size', scope: 5, whitelist: ['14']}), true, true
     )
 
-    // eslint-disable-next-line @typescript-eslint/dot-notation
+
```
Comment (KillerCodeMonkey) [inline projects/ngx-quill/src/lib/quill-editor.component.spec.ts:1358]: lines with only white spaces / trailing whitespaces should not be allowed the, too

**Comment by KillerCodeMonkey** [comment]
oh i never noticed, cause my code editor already is linting it on the fly

File: projects/ngx-quill/src/lib/quill-editor.component.spec.ts (hunk 1 above)
Comment (BBboy01) [reply projects/ngx-quill/src/lib/quill-editor.component.spec.ts]: yes, just run `eslint --fix`, seems we need more `@stylistic/eslint-plugin` rules

Comment (KillerCodeMonkey) [reply projects/ngx-quill/src/lib/quill-editor.component.spec.ts]: i think you already know, but maybe https://github.com/angular-eslint/angular-eslint/tree/main/packages/angular-eslint/src/configs is helping

Comment (BBboy01) [reply projects/ngx-quill/src/lib/quill-editor.component.spec.ts]: demo of angular-eslint merged ts rules and angular-ts rules, if you pefer merge them, we can handle it too

**Comment by BBboy01** [comment]
I noticed there is a `.prettirerc` file but no prettier as dependency, I'm not sure are you using prettier for code formatting

we can add prettier as lint rules too

**Comment by KillerCodeMonkey** [comment]
> I noticed there is a `.prettirerc` file but no prettier as dependency, I'm not sure are you using prettier for code formatting
> 
> we can add prettier as lint rules too

i guess when using stylistic it is not a good thing to add prettier as well. they are more or less there for the same purpose.

**Comment by BBboy01** [comment]
> i guess when using stylistic it is not a good thing to add prettier as well. they are more or less there for the same purpose.

Oh right, fair enough, leave it just as it is now

**Comment by KillerCodeMonkey** [comment]
can you check the ci and ping me, if you think the pr is ready?

PS: nice to see/have a real life example with angular + stylistic. In other projects i hate to touch those things...

**Comment by BBboy01** [comment]
@KillerCodeMonkey it's ok now, seems like my global npm proxy caused the error

**Review by KillerCodeMonkey — APPROVED** [review]

**Comment by KillerCodeMonkey** [comment]
@BBboy01 thanks :+1:

**Comment by BBboy01** [comment]
https://ngxtension.netlify.app/utilities/migrations/inject-migration/

if you like, we can easily use ngxtenstion provided script to migrate angular decorators to signals

**Comment by KillerCodeMonkey** [comment]
currently for inputs input signals are used. i am waiting til "model" signals are stable and replace some input with model signals.

for outputs some more refactorings need be done, cause we are adding and triggering some outputs only if there are observers. So i need to check, if we need this logic at all and if, how we can adapt it with signals.

**Pull request merged**